### DISCUSSION TITLE
Javascript Condition in Exclusive Gateway

### DISCUSSION THREAD

**Discussion by MaheshkumarSundaram** [bpmnServer/bpmn-server#125] [General]
Hi @ralphhanna 

_`Using v1.3.28`_

When including conditions for the exclusive gateway, I need a clarification.

The example condition I am using:

```js
item.data.arrObj.filter(e => e.key1 === 'dw' && !e.key2.includes('SAR')).length > 1
```

This works without any problem.

But if I do this:

```js
item.data.arrObj.reduce((accumulator, e) => {
    if (e.key1 === 'dw' && !e.key2.includes('SAR')) {
      accumulator++;
    }
    return accumulator;
  }, 0) > 1;
```
The one with `reduce` doesn't work. I am not sure why?

Any help? Thank you.

**Comment by ralphhanna** [comment]
sorry this one fell through the cracks!!! , is it still an issue?

**Reply by MaheshkumarSundaram** [reply]
Tbh, I haven't revisited this too. I'll double check and get back to you, Sorry.

**Comment by MaheshkumarSundaram** [comment]
@ralphhanna, Tested it. Not an issue anymore. Thanks.
